### ISSUE TITLE
There are result percentages being tallied to totals, without results or collected evidence

### ISSUE THREAD

**Issue by RickieT** [mercycorps/TolaActivity#1065]
See attached image:
![release_block](https://user-images.githubusercontent.com/23220180/51149993-4a38db80-1819-11e9-8b15-73ffabb9f378.png)

**Comment by jennyhavah** [comment]
This may be connected as well: We are missing this warning:

![image](https://user-images.githubusercontent.com/33670923/51150232-45285c00-181a-11e9-9a87-6fda94a1096f.png)

**Comment by jennyhavah** [comment]
We have other broken warnings that may be related to this issue or possibly need to be in a separate ticket.

To reproduce:

1. Add an indicator with Midline/endline targets.
2. Add a result.
3. Now delete all your targets and don't create new ones.

Here's what I see:
![image](https://user-images.githubusercontent.com/33670923/51150393-f7f8ba00-181a-11e9-8329-1a927c40b498.png)

Here's what I'd expect to see:
![image](https://user-images.githubusercontent.com/33670923/51150455-3bebbf00-181b-11e9-8265-a7875a588954.png)

**Comment by jennyhavah** [comment]
One more... I then re-added my targets to the above example.
![image](https://user-images.githubusercontent.com/33670923/51150542-9b49cf00-181b-11e9-9cd9-de94b00242ba.png)

Here's what I'd expect to see (from demo)
![image](https://user-images.githubusercontent.com/33670923/51150507-6fc6e480-181b-11e9-9ac4-0221ced82876.png)

**Comment by axoplasm** [comment]
@jennyhavah & @RickieT how can I recreate the bug as originally noted by Rickie in the first screenshot?

**Comment by jennyhavah** [comment]
> how can I replicate the bug as originally noted by Rickie in the first screenshot?

@axoplasm My understanding is that this error occurs when a result either has no date associated with it or has a date outside the program period. We put a bunch of validation in place to prevent this error going forward -- making the date a required field and enforcing dates inside the program period -- but we have older results in the system that still need to be accounted for.

TL;DR I think you'd need to go into the database and try deleting the date associated with a result, or entering a date outside the program period.

**Comment by jennyhavah** [comment]
Wondering what's going on here -- this is the BRESHNA program from Afghanistan, indicator 1.3.1.

## Production
![image](https://user-images.githubusercontent.com/33670923/51227211-21434400-1908-11e9-9f96-774e5cb87c57.png)

## QA
![image](https://user-images.githubusercontent.com/33670923/51227231-328c5080-1908-11e9-88ba-e4f813d3d26c.png)

In the QA version, there is no collected data, and yet there is a LoP Actual of 5. How is that possible?

**Comment by KennyMonster** [comment]
> In the QA version, there is no collected data, and yet there is a LoP Actual of 5. How is that possible?

The DB is in an inconsistent state, and probably has been for some time.

The indicator **1.3.1 Number of EPC...** has an id of 4845.

There is 1 result that references indicator id 4845, which also references periodic target with ID 4460.

Periodic target with id 4460 references indicator id 4840.

**4840 != 4845**

It's unclear why a result references indicator 4845 but a periodic target attached to a different indicator with 4840. The creation date of the result is 2018-02-08 10:58:48 which means this inconsistency has probably been around for awhile, but was hidden from view due to how previous queries were performed.

Unsure how to proceed as this is a data problem more so than a code problem.

**Comment by KennyMonster** [comment]
All results in an inconsistent state (only 6)

[inconsistent_results.csv.zip](https://github.com/mercycorps/TolaActivity/files/2770368/inconsistent_results.csv.zip)

**Comment by RickieT** [comment]
So issues attached unrelated to the issue described in this ticket are not reproduceable on QA currently. I am also unable to reproduce the main issue in this ticket, so I will consider this issue resolved
